.github/workflows/pre-submit.lint.yml

```diff
@@ -0,0 +1,79 @@
+name: Lint
+
+on: [pull_request]
+
+permissions:
+  # Needed to check out the repo.
+  contents: read
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      - uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923 # v3.2.0
+        with:
+          go-version: "1.18"
+      - env:
+          GOLANGCI_LINT_VERSION: "1.46.2"
+        run: |
+          set -euo pipefail
+
+          #Install golangci-lint
+          wget -qO- "https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz" | tar -xzf -
+          mv golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64/golangci-lint /usr/local/bin
+
+          # Run golangci-lint
+          golangci-lint run ./... --out-format=github-actions --timeout=5m
+
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      - env:
+          SHELLCHECK_VERSION: "0.8.0"
+        run: |
+          set -euo pipefail
```

Comment:
Why not use an action like this https://github.com/ludeeus/action-shellcheck?

It gets all the benefits of upgrades and maintenance.

Reply:
Partly for simplicity (I'm removing reviewdog because it had bugs) and partly to reduce dependencies on external actions.

Reply:
I understand if there are dependencies on the `library` code. But these are tools.  What are the concerns of having dependencies on tools?

Reply:
In this case it's not so bad because pull_request events have fewer permissions but we do need to be concerned about running 3rd party code in a context where it has permission to do things on the repo.

Reply:
What are the specific concerns?

Can you please be specific?

It would be helpful to understand. 

As long as the repository has the beam protection turned on and of the permissions have been set to least privileged it shouldn’t be an issue. 

Also AFAIK these actions don’t need any write permissions to the repository.

Reply:
My reasoning is:
1. The backstory for why this PR was created in the first place was because 3rd party actions (specifically reviewdog) had bugs that allowed pre-submits to pass even though there were issues on PRs. Debugging was very painful and time consuming, and I want us to have simple pre-submits that are understandable by anyone, and we able to fix bugs in pre-submit ourselves.
2. I would rather have a blanket policy of not using 3rd party actions rather than a policy which excludes the pull_request event because it's more likely to be broken later if we add new triggers to the workflow (push, or pull_request_target for example).
3. I think we need to limit the use of 3rd party dependencies when possible as a general rule. My overall view is that we need to be fairly conservative about how we use 3rd party dependencies and actively work to minimize them given that this repository hosts the reusable workflows that will be called by other projects. Esp. after we have more adoption, a compromise or incident on this repo would be a fairly high-impact event.

My benefit/analysis of using our own code:

Pros:
- presubmits are simple and are easier to debug
- 3rd party actions aren't used
  - They never get permissions on the repo (regardless of event)
  - We don't need to audit the code on the actions themselves

Cons:
- dependabot doesn't update the versions of the linters

In my view the pros outweigh the cons.